### ISSUE TITLE
V 0.39 

### ISSUE THREAD

**Issue by pinkwolf6686** [DxCx/plugin.video.9anime#267]
Just updated the latest version of 9Anime and it is still saying no eligible sources.  What is going on with this site.  I hope that it gets fixed have been trying for the past 2 days now.

**Comment by SwagOtaku** [comment]
This time the 9anime server is down, no episodes working on the site. Just got to wait it out

**Comment by ghettowboy** [comment]
Yeah, seems like they had too much fun playing around that they broke it. Lol

**Comment by somedudenamedrush** [comment]
it seems as if they changed the domain slightly from www5.9anime.is, to www6.9anime.is,

**Comment by WonderfulSubs** [comment]
@somedudenamedrush nah that's just called load balancing. Nothing to do with the issue

**Comment by DxCx** [comment]
Yeah.. they probably changed something again..
Will look into it when i have more time..

**Comment by Karma32** [comment]
it seems like 9anime is constantly changing their tokens. I took a look at the java script and they've added a function that gets the date which is then used to make a new token on a daily basis

Hope this helps

**Comment by sd-92** [comment]
Any updates on this??

**Comment by ghettowboy** [comment]
People need to be patient. If it's true that they are changing it daily, that might be an issue.

**Comment by Peingod300** [comment]
I'm totally fine with being patient, I just also want to be updated and make sure it wasn't something I screwed up :) Hopefully this gets a good fix soon as possible :)

**Comment by username09090** [comment]
If the token is being changed daily does that mean the DD = 'X8uEFlj2' in NineAnimeUrlExtender needs to be changed daily. And if so how to u get that code or maybe someone can start a comment chain with the updated daily token. I don't mind changing 8 letters of code each day for now.

**Comment by RedNinjaX** [comment]
I believe I've found a resolution to this that doesn't require having to constantly update the DD token and other parts of the code, however I'm still working out some kinks. I'm a JavaScript guy though but the code could easily be rewritten in Python (as I did the same with the NineAnimeUrlExtender code).

**Comment by pjcanales** [comment]
Good work.  Progress is a path to success

**Comment by DxCx** [comment]
@redninjax feel free to send me it in JS ill recode over the weekend..
Unfortunately i dont really have time to do it till then..

**Comment by DxCx** [comment]
Also if there will be no choice then ill export it as settings..

**Comment by RedNinjaX** [comment]
@DxCx will do 👌

**Comment by grabberboy** [comment]
@RedNinjaX send me the code ill convert it to python if @DxCx is busy
i have a free time now

**Comment by RedNinjaX** [comment]
@grabberboy sure. I should have something by Friday or earlier.

**Comment by FireWolf74** [comment]
I installed the new version a couple days ago and everything worked.

**Comment by sd-92** [comment]
@FireWolf74 It briefly worked until 9anime changed their code again. I suppose it is not working for you at the moment?

**Comment by ghettowboy** [comment]
Did anyone actually think 9anime supported kodi in the first place? So why would his statement shock anyone.

**Comment by RedNinjaX** [comment]
It doesn't. Just found it funny that he just so happened to respond to that as recently as the code broke.

**Comment by ghettowboy** [comment]
It is funny, because people were probably hounding him on Reddit, and on Discord.

**Comment by romxuk** [comment]
Does anyone have the most current working token or has nobody worked it out yet?

**Comment by Prometheusx-git** [comment]
a29856fa

**Comment by RedNinjaX** [comment]
"a29856fa" is definitely it...for the next 2 hours or so

**Comment by Karma32** [comment]
i'd love to have look at how you generate the tokens any pointers on how to do it

**Comment by RedNinjaX** [comment]
@Karma32 I finally figured out how he gets it ;)

![screenshot from 2018-06-21 14-07-29](https://user-images.githubusercontent.com/34778154/41737177-7a7e5924-755c-11e8-8928-c59205356bf8.png)

**Comment by romxuk** [comment]
what js file did you find that in karma32? I am interested in seeing if i can auto pull new tokens from it

**Comment by romxuk** [comment]
found it myself in all.js

**Comment by bashulsbosch** [comment]
@DxCx, thanks for everything dude, you're awesome!

**Comment by Randomdude707** [comment]
Has the sources issue been resolved yet

**Comment by JustAguyWhoLikesAnime** [comment]
@Randomdude707 no but like @RedNinjaX said he may have found a resolution and may be done by Friday

**Comment by romxuk** [comment]
made a test script to pull tokens easily for me, if it works after next token change I will share a link to get keys as they change

**Comment by DxCx** [comment]
Or we can just integrate it into the addon ;)

**Comment by romxuk** [comment]
Not a problem if my method works I will happily share but I am not sure it does until they change the token again, If my simple script does work I will let you know how I did it. test page (nothing but the token) which I wrote in php http://107.150.36.26/9anime.php if it changes to the right token when they update I will let you know how I got it

**Comment by RedNinjaX** [comment]
I was going to go about it in a different way

For instance instead of worrying about coming up with a script that will generate a new token on the fly, I'm making a script that will be able to take the functions of all.js and rework it to generate the extra parameter without having to worry about any other major changes to the script (like the one from a recent update where you had to change those few parts of code from using zero to using the current iteration loop count). Instead of worrying about the DD token alone, no more having to worry about the extra parameter script as a whole. Also keep in mind that your token generator script will potentially be null and void within a few hours since all.js itself is changing every few hours (you can keep a tab on the versions by looking at the text after the question i.e. all.js?abc123)

**Comment by HatHeadNinja** [comment]
@RedNinjaX 9anime seems a bit knee jerk about it. People who can install and maintain Kodi require a certain tech savvy and likely have browser plug-ins to defeat 9anime's aggressive adverts.  It would also seem that retrieving data for lists on Kodi would be lighter on their servers than serving all those images and other assets via their website.

**Comment by RedNinjaX** [comment]
@HatHeadNinja you're absolutely right. Although most of the image assets on 9anime are actually hosted on third party sites/CDNs themselves, meaning the only other load they're getting from users visiting their site directly is from the extra http requests that the site makes for self-hosted js files and their own API requests such as the ones to get the embed links, meaning they don't even have as much of a strain as they try to make out (even though I'm sure there's still plenty of server load).

In the end it all boils down to loss of ad revenue

**Comment by RedNinjaX** [comment]
Yeah just as I predicted: new all.js revision after a few hours with a new way to generate the DD var breaking any previous methods of generating a new one. This can only be solved 1 of 2 ways (or 2 1/20

Anyway current new DD is: 49220519

**Comment by JustAguyWhoLikesAnime** [comment]
will that change what you are planning @RedNinjaX to fix this so it will not be a problem?  I am looking forward to see how you are able to make this work :)

**Comment by grabberboy** [comment]
it seems theres no way out to fixed this yet since the new all.js all variables are auto changing after some few hours.
i hope @DxCx can fix this @Prometheusx-git  @hoppler 
now i cant watch anything until this kodi works again.
should i hired at freelancer.com?
im willing to pay so i can help this community.

**Comment by jeremyjpj0916** [comment]
In the mean time another okay app I found was Dubbed Anime, less source choice but all good working links. Look it up i think its on blamo repo

**Comment by JustAguyWhoLikesAnime** [comment]
@RedNinjaX nice looking forward to it :) since the update never seems to work for me so I have to update it the old fashion way so this would be handy ha

**Comment by Karma32** [comment]
@RedNinjaX where did you find the new DD i looked through the java script and couldn't find

**Comment by RedNinjaX** [comment]
I'd tell you but it changed again. The function itself always changes as well as the variables

Current DD: **cea7cb61**

**Comment by HackedGameMaker** [comment]
I'm curious how's the issue coming along?

**Comment by JustAguyWhoLikesAnime** [comment]
RedNinjaX says he will be done today or Saturday as the latest @HackedGameMaker

**Comment by dgaubatz313** [comment]
I use 9Anime as a backup. I usually watch anime through VRV, which only cost $10 a month and gives you access to Crunchy Roll and Funimation as well as other things.

**Comment by RedNinjaX** [comment]
@dgaubatz313 now what if I told you you could could have both VRV and 9anime rolled up into one at the same "monetary" convenience as 9anime and other similar streaming sites ;)

Anyway @HackedGameMaker personally I'm still in the middle of developing the solution. So far so good with the progress. I don't know how anyone else is doing though

**Comment by dgaubatz313** [comment]
@RedNinjaX That sounds nice :) If everyone did it though, we would have no anime.

**Comment by ghettowboy** [comment]
I pay for CR but there is A LOT of anime  that Funi and CR just plain out ignore. That's why I love 9anime and gogoanime...

**Comment by dgaubatz313** [comment]
@ghettowboy Some shows they don't have in dubbed either when the dubbed versions are out. Maybe if they were as big as Hulu that wouldn't be the case. I don't think they'll ever be as big as Netflix.

**Comment by d1arkstar** [comment]
@RedNinjaX thanks for the DD is seem to be working still. Also how did you figure out the  tok ?

**Comment by Jkrise** [comment]
So this may be a stupid question and I apologise if it were answered else where, but how does one update the DD? 
Would it be possible to add a menu option to allow the users to manually update or set the DD?
That way if it is wrong for any reason you just open the menu and type in the new one once you find it.
Or, would that be too difficult to add/code?

**Comment by d1arkstar** [comment]
if you are on pc you got to the file go to NineAnimeUrlExtender.py
open it and look for DD = then replace what it in the ' ' quots wilth 
cea7cb61

but when your on this page click on code then  click fix: NineAnimeExtender Token (By Prometheusx-git) 
and you see where it needs to get updated

**Comment by d1arkstar** [comment]
also make a copy of NineAnimeUrlExtender.py before changing it

**Comment by ghettowboy** [comment]
@dgaubatz313 Hulu and Netflix are hardly giants, well known frontrunners yeah. Also Funi and CR are more like partners now since they agreed to split the subs(CR)/dubs(Funi) between them to lighten the load. But I always find great shows that get missed through alternative sources.

**Comment by d1arkstar** [comment]
@Jkrise  adding that option would kill the app it would scare people in think know I have to go hunt down the token and cause issues

**Comment by RedNinjaX** [comment]
Guys just hang on tight. I'm almost done 😉

**Comment by Randomdude707** [comment]
Will do 😁

**Comment by Randomdude707** [comment]
I kind of feel bad since I don't know how to do any of this stuff or help but hey I commend you for being able to do pretty much any of this 🖒

**Comment by Jkrise** [comment]
@d1arkstar, thanks for the info, I wasn't sure and it makes sense, I was just trying to think of a way to make it easier on people.

@RedNinjaX, thank you; no way I would be able to do that

**Comment by JustAguyWhoLikesAnime** [comment]
@Randomdude707 ik what you mean

**Comment by MobileSuitSadBoi** [comment]
When the solution gets finished can you please provide instructions on how to install for the firestick! I kinda new to the scene so it's a little confusing sometimes. Thanks!

**Comment by HackedGameMaker** [comment]
@MobileSuitSadBoi In order to install it on your fire stick you first need to have kod i installed.

**Comment by xSLOWWx** [comment]
So is the issue fixed?
Did @RedNinjaX fix it?

**Comment by MobileSuitSadBoi** [comment]
I have Kodi installed already lmao, I download DxCx v38 but couldn't find sources. I just instructions on where to find the solution when it's done, I can figure my way out from there lol.

**Comment by HackedGameMaker** [comment]
If you read his readme then it should explain everything for you, if you've already downloaded it to your fire stick then go click on add ons then once you're in the add ons tab go to where it shows a box that looks open and the go to repo and look for 9anime and click on auto update then when the plugin gets fixed and they update the app it should all be go to go.

**Comment by ghettowboy** [comment]
@JustAguyWhoLikesAnime I have that issue also, what I've been doing is going into System>addons>repositories select 9anime repo and check for updates. That's been working for me...

**Comment by JustAguyWhoLikesAnime** [comment]
been clicking on the app and clicking check for updates but it never works ill try your way when we get an update thanks @ghettowboy

**Comment by FireWolf74** [comment]
Ok Boys and Girls
Time to relax a little.

https://www.youtube.com/watch?v=2VfWB3_VzcY

**Comment by HackedGameMaker** [comment]
@JustAguyWhoLikesAnime I think you're pressing the wrong check for updates, once you're in the section that says install from repository and install from zip if you're on a fire tv then I do believe you just have to press left and it should being up the options menu and that's where you'll find check for updates and you should get the issue fixed, I'm not sure if you've done this already but if you have then I'll just leave this here for anyone else who hasn't figured this out.

**Comment by JustAguyWhoLikesAnime** [comment]
@HackedGameMaker  ill give it a try when the thing gets a update idk why the normal one does not work though lol

**Comment by username09090** [comment]
Where are u finding the DD token again? So I go onto 9anime website, do I need to start an episode to get the token, or can i pull it from the home page? Next which all.js file is it in? In chrome i go to sources than where? assets > ?

**Comment by RedNinjaX** [comment]
@username09090 finding and updating the DD token manually is pointless now as it updates every few hours now. I'm almost finished with the script. It will definitely be out by today.

**Comment by JustAguyWhoLikesAnime** [comment]
@RedNinjaX like I said brfotr your the man lol. I been so bored

**Comment by RedNinjaX** [comment]
@DxCx @grabberboy 

I'm happy to say that I'm done! Note a few things:

- The code only returns the DD token, so therefore if there are any changes to the rest of the code that generates the extra parameter, those changes will have to be made manually. I'm working for an automatic solution to that as well, but considering that portion of the code has only changed once over the last few revisions, it's probably nothing much to worry about.

- I've tested it with all of the all.js revisions so far and it has worked but I can't guarantee that there won't be some kind of future revision that will break it, therefore needing myself or others to update it again but it should be a lot more simple then as it is just a matter of finding the common position of the token variable again.

- I'd recommending running this is an API server side since that way any 9a snoops can't see how it's done.

- Written in JS via Node.js. Feel free to port it over to Python if you can. If not, there is always the server side option using something like Express.js to host it.

-ZIP file with code attached

[DD.zip](https://github.com/DxCx/plugin.video.9anime/files/2130230/DD.zip)

**Comment by MobileSuitSadBoi** [comment]
Ok so do I download the zip you posted and 9a should work? Or are there other steps I have to take

**Comment by MobileSuitSadBoi** [comment]
Roger that @JustAguyWhoLikesAnime

**Comment by Karma32** [comment]
@RedNinjaX so is the latest token bfcad671 guess i want to try something

**Comment by Randomdude707** [comment]
Hooray it's fixed awesome job everyone 🎉🎉🎉🖒🖒🖒

**Comment by grabberboy** [comment]
@RedNinjaX 
GREAT!
hoping @DxCx will have a time asap!
converting JS to python is very possible.

**Comment by vn-ki** [comment]
@grabberboy I took a look at the code and I don't think it'll be that easy. @RedNinjaX s code runs actual JavaScript code (or so I think). 

The best way I can think of doing this is by running the js code in node and capturing the output using python. I think this will not be possible with a Kodi plug-in.

The other suggestion that someone run a server with the code will be the best, I think.

**Comment by RedNinjaX** [comment]
@vn-ki which is why I recommending hosting the script on a server above all else. Hook it up to an Express server as an endpoint and simply send a request to the endpoint when a new token is need it (presumably when 9a's API returns an error in the app) and then save the new token in a file in the app.

**Comment by romxuk** [comment]
I have a few dedicated servers I am not using much right now, If need be I could let it run on one of them although they are fairly basic servers but should be fine for the usage this needs, one is 100mbit and the other is 1gbps

**Comment by RedNinjaX** [comment]
@grabberboy in that case if you want you want to avoid hosting (and yes I absolutely agree with you on that note, but it feels like the simplest choice if you're having conversation issues) then to properly convert my code you need to find a python equivalent of the JSDOM Node.js package, then you'll be fine. It would be way more efficient than emulating a browser upon each video request as that's too slow (plus JSDOM doesn't emulate the entire browser, just the DOM to mimic the DOM window and document so it processes code fast, unless you choose to use the VM aspects of it)

Otherwise the only other solution to your server hosted worry is to use a proxy on each request and we'll...trust me that's slow

**Comment by xSLOWWx** [comment]
So when is this thing getting fixed?
Gotta watch seven deadly sins on my TV 😉

**Comment by JustAguyWhoLikesAnime** [comment]
@xSLOWWx we do not know it is going to be looked at tomorrow

**Comment by romxuk** [comment]
I am sure one of the options is being worked on, we just need to sit tight and be patient. It takes time to weigh up options and put something completely new in and re write script from one language to another

**Comment by RedNinjaX** [comment]
@romxuk that's the spirit :)

**Comment by Jkrise** [comment]
I too can't wait, but I lack the skill set need to be of any real help. I appreciate all of the work you guys do. And while I really wish it would get done, like yesterday, I am eagerly waiting as patiently as possible, (though I am quickly running out of finger nails!)

**Comment by strongstyle** [comment]
I just registered to say redninja is the man

**Comment by RedNinjaX** [comment]
@strongstyle Awww that's so sweet ^-^ Thanks a lot!

Also I just want to note that to anybody manually doing updates to the DD token and extra parameter code, you have to change the "n += i" parts to "n *= i" in order for the new tokens to work. Since the all.js revisions seem to be making more frequent modifications to the actual extra parameter than before it's becoming more necessary to figure out a way to permanently extract the function out of each iteration that generates this param. I _think_ I'm close to a solution...

**Comment by DxCx** [comment]
well i've looken into this @RedNinjaX and thats emulated JS..
i've from almost a year ago a much better solution of emulated JS to just extract stuff out..
then i thought of chaing the plugin to work against an endpoint and opened dontations to see if people will support further developemnt..
since then no body donated and i just put that aside..
i've tried to push kodi into emulate JS and thats really much of an hassle,
to run emulated JS we will need a middle endpoint between the site and the addon which i cannot effort to be paying right now unfortunately..
if anyone can maintain such "proxy" service and want to, lets discusses
this will also allow the plugin to work with kodi version < 17

**Comment by RedNinjaX** [comment]
@DxCx @vn-ki @grabberboy I have a proposal. In that case I believe that the best and only proper solution is this:

- I've figured out a way to get the entire URL to be generated without having to worry about grabbing a new token or modify the extra parameter code. Instead, just pass in the ID of the episode, the TS, and the server number to the function and it will return the full URL needed to grab the embed link.
- This still uses the emulated JS method, or I believe the proper term would be emulated DOM, to generate the URL however instead it would work like this:
  • Once per hour, a script would automatically fetch the latest all.js revision as well as an html copy of the 9a home page and save them both into their own respective files. The request can be done via a proxy if needed, considering this is done only once per hour and the scripts are saved to the hard drive, there is no need to worry about a slow request whenever an app, plugin, or whatever calls for one
  • With this, using Express.js, users can make a request via an endpoint with params passed and within a few seconds get a resulting json with the embed link (note when I say "users" I just mean applications)
  • If an hour hadn't passed yet and all.js has changed and an app receives an error as a response, a special param (I guess retry=true) can be used to try again, and this time this forces the script on the server update it's copies of all.js and the 9a home page html which will update the proper code for generating the URL
  • Since JSDOM only emulates the DOM and not and an entire browser (like PhantomJS does) its quick to run, and without the extra time spent making the requests to all.js and the 9a home page like my current DD token generator does, it will process requests at an event faster rate.
  • If a subset of us managed this server, we can either make it available for all 3rd party app developers to use or only a select amount if we choose. This also prevents others (perhaps even a certain "opposer") from being able to see how the code works.
  • Since the script uses a relatively low amount of RAM, an inexpensive VPS is all that is needed. I recommend Vultr as that's what I use for mine. They have a plan starting at just $2.50 a month, meaning only $30 a year needed to keep the server running. If some of us are willing to pitch in, it only helps

And on top of that like you said, you can make the app compatible with more versions of Kodi this way.

Thoughts and/or opinions?

**Comment by DxCx** [comment]
`I've figured out a way to get the entire URL to be generated without having to worry about grabbing a new token or modify the extra parameter code. Instead, just pass in the ID of the episode, the TS, and the server number to the function and it will return the full URL needed to grab the embed link.`
is basiclly what i had and wanted to push forward..just only for URL then i had 1 request to get the TS and back then there was no server number..
i think that having a middle backend is the way
also, using dataloader we can easily cache requests
so if for example a new episode is out and alot of people fetch it at once, it will make much less requests to get the episode

**Comment by DxCx** [comment]
btw regarding costing, i really don't know, maybe running a lambda is the right solution? (i've heard it should be the cheapest)
im really not good with those so if someone can advise that can be awsome

**Comment by ruegnueg** [comment]
actually in the meantime I cast it from tablet to kodi or TV or firetv directly. works quite well. but I'm not a friend of casting 😉

**Comment by chrissyt29** [comment]
Will it be fixed for Kodi?

**Comment by disbjon** [comment]
Made a account to just thank @DxCx and @RedNinjaX and everyone involved to get the add on working again. I appreciate all the work involved to make the add on, keep it running and sharing it us the fans!

**Comment by RedNinjaX** [comment]
@disbjon thank you!

And @DxCx I don't know about AWS and their rate of charge but if it keeps the plugin going as well as my own anime project and other 3rd party apps, then I'll be willing to pay the $30 a year for the server and maintain it.

**Comment by romxuk** [comment]
As stated, if you want to try it using a server let me know I can set something up for you for free to try it out as I have a few servers not really being utilized and some spare IP's so I can give 1 or 2 ips and setup a VPS  (not restricted, but just so it is in its own personal space and does not affect my usage on the server and mine does not affect yours). If that is of use let me know and preference on OS and if you need Apache and MySQL etc installing.

**Comment by JustAguyWhoLikesAnime** [comment]
sounds like this will be fixed any day now :) I am happy I been missing my anime ha

**Comment by nateclub** [comment]
Yo great stuff guys!!!

**Comment by MobileSuitSadBoi** [comment]
I CANT WAIT FOOD WARS NEW EPISODE IS OUT AND I NEED IT

**Comment by xSLOWWx** [comment]
What is an estimated time that it will be up and running.

**Comment by RedNinjaX** [comment]
@DxCx sure, will do 👌

@romxuk if you're serious about that then that would be extremely appreciated! Preferably Ubuntu 16.04 for the OS if you could and preferably with LEMP preinstalled

**Comment by dayv6880** [comment]
I know you probably are all super dedicated to using 9anime but in the meantime while it is down you could use WatchNixtoons by Nixgates.

**Comment by xSLOWWx** [comment]
@dayv6880 yeah just downloaded that repository and it's trash compare to this one. 
Need this one fix fast!!

**Comment by romxuk** [comment]
@RedNinjaX just installing VPS things now and will build a VPS on Ubunto 16.04, obviously need to sort something about username and password out, I will also install LEMP to it. anything else before I get started?

**Comment by Randomdude707** [comment]
You guys could use an apk called anYme while this one is down although it's meant for phones and tablets it works on other devices however if your using a firestick (like I am) you will need another apk which is mouse toggle hope this helps those looking for a way to watch anime and keep up the great work guys.

**Comment by disbjon** [comment]
For fire sticks download “Silk Browser” from the store and go to 9anime.is web site directly for a quick fix.

**Comment by romxuk** [comment]
@RedNinjaX done and installed LEMP, do you need me to secure MySQL using same pass as server (obviously can be changed) and do you need phpmyadmin installing to manage any databases? or are you OK doing any other things that are needed.

**Comment by kasei-ibara** [comment]
Could someone explain to me how to find the DD token from the website, please? I don't mind manually updating it until the fix is done, but I've looked in the all.js files for a few 9anime pages and couldn't find anything that looked like it...

**Comment by RedNinjaX** [comment]
@romxuk awesome 👌 and yeah please secure MySQL using the same password. As for phpmyadmin you can just go ahead and install it considering you already have MySQL installed anyway. Anything I can deal with myself. Thank you so much!

**Comment by RedNinjaX** [comment]
@kasei-ibara there's really no clear cut answer to that anymore as the location and function names and variables keep changing, but using DevTools you can track down the location. Usually it's around a few variables "var n,e=0" in a function that has the parameters "(t, I)" in it. You can add breakpoints in DevTools to track it down. If it helps the current DD token is **bfcad671**. So far it hasn't changed since yesterday.

**Comment by JustAguyWhoLikesAnime** [comment]
@RedNinjaX  man you guys are getting me hyped ha

**Comment by romxuk** [comment]
@RedNinjaX done, rest needs discussing non public like passwords etc. 3:30am here sorry for slow timing sorting it but had a lot of work to do as well. let me know how to contact you direct and I can give you the credentials

**Comment by RedNinjaX** [comment]
@romxuk it's alright no worries and great. Do you have a Discord account? If so my Discord tag is **RedNinjaX#0722**

**Comment by HackedGameMaker** [comment]
So how close is the issue from getting fixed? I wished I could've help everyone involved but since I haven't coded in a while I'd probably make things worse so I guess I'll just keep waiting for you guy's to go all out for all of us that can't help you out with this.

**Comment by RedNinjaX** [comment]
@HackedGameMaker we've got it under control 😉

**Comment by HackedGameMaker** [comment]
@RedNinjaX  I know I got faith in all you guys

**Comment by DxCx** [comment]
Yes. We are working togather to solve this.
Hope that by next week everything goes back to normal

**Comment by disconsented** [comment]
Could we get a separate issue that details the problem without needing to wade through everything unrelated?

**Comment by romxuk** [comment]
@RedNinjaX sent you a request on discord

**Comment by romxuk** [comment]
@disconsented it is 1 issue and that is the DD code needed is changing regular now meaning a solution to automatically update the DD has to be created. A solution should be ready soon so another issue would not really be useful as the problem is known and being worked on as urgently as possible. We should all be able to get back to watching the best anime addon on Kodi very soon

**Comment by RedNinjaX** [comment]
@romxuk friend request accepted

(Also not just the DD code but also the rest of the code that generators the extra parameter code is also changing every now and then too but not as frequently)

**Comment by romxuk** [comment]
@RedNinjaX info sent, if you have any issues let me know

**Comment by RedNinjaX** [comment]
@romxuk awesome. Just messaged you back

**Comment by xSLOWWx** [comment]
So we are estimating the repair will be done by next week?

**Comment by romxuk** [comment]
@xSLOWWx it could be done much sooner, There should be an update soon on how long it might take, @RedNinjaX and @DxCx are working on it.

**Comment by strongstyle** [comment]
Hey ninja, sent you a request on Discord.

**Comment by dayv6880** [comment]
I guess you guys working on it now has the whole add-on not working at all now.

**Comment by epicshot01** [comment]
@dayv6880 that might be because 9anime has changed from www5.9anime.is to www6.9anime.is

**Comment by dayv6880** [comment]
No it was just my fire stick being stupid

**Comment by RedNinjaX** [comment]
@DxCx I would recommend just updating the token in the meantime as it hasn't changed in 3 days so who knows, this whole thing might be over already. The current token is **bfcad671**

**Comment by vn-ki** [comment]
Yeah! @RedNinjaX I noticed it too. @DxCx You can just merge @Prometheusx-git's PR. It works!

**Comment by romxuk** [comment]
that DD code isnt working for me am I missing some other change or something?

**Comment by RedNinjaX** [comment]
@romxuk change both **n +=** to **n *=** in the extra parameter code. I believe there is another change as well both first try that out.

**Comment by romxuk** [comment]
@RedNinjaX  working now thx

**Comment by Jkrise** [comment]
I just downloaded 0.40 but I still get the same error about no eligible sources. An i doing some thing wrong?

**Comment by RedNinjaX** [comment]
@Jkrise #282
